src/Symfony/Component/HttpKernel/Controller/TraceableControllerResolver.php

```diff
@@ -52,12 +59,20 @@ public function getController(Request $request)
 
     /**
      * {@inheritdoc}
+     *
+     * @deprecated This method is deprecated as of 3.1 and will be removed in 4.0.
      */
     public function getArguments(Request $request, $controller)
     {
+        @trigger_error(sprintf('The %s method is deprecated as of 3.1 and will be removed in 4.0. Please use the %s instead.', __METHOD__, TraceableArgumentResolver::class), E_USER_DEPRECATED);
+
+        if ($this->argumentResolver instanceof TraceableArgumentResolver) {
```

Comment:
That's sadly incompatible because the `ControllerResolverInterface` doesn't have to have the `ArgumentResolverInterface` and in that case I would still have to add the stopwatch manually.